[UI] ui.test_variables/SmartVariablesTestCase/test_positive_update_*

The value update seems to went wrong

```
args = (<tests.foreman.ui.test_variables.SmartVariablesTestCase testMethod=test_positive_update_type>,)
kwargs = {}, settings_project = 'sat'

    @wraps(func)
    def wrapper(*args, **kwargs):
        """Wrapper that will skip the test if the test method project does
                not match with the settings project.
                """
        # Validate project value
        if settings.project:
            settings_project = settings.project.lower()
        else:
            settings_project = 'sat'
    
        if project not in allowed_project_modes:
            raise ProjectModeError(
                '"{0}" is not a project mode. The allowed project modes '
                'are: {1}'.format(project, allowed_project_modes)
            )
    
        # If robottelo.properties not present or does not specify a project
        # use sat
        if settings_project not in allowed_project_modes:
            raise ProjectModeError(
                '"{0}" is not an acceptable "[robottelo] project" value '
                'in robottelo.properties file. The allowed project modes '
                'are: {1}'.format(settings_project, allowed_project_modes)
            )
    
        # Preconditions PASS.  Now skip the test if modes does not match
        if project != settings_project:
            raise unittest2.SkipTest(
                'Server runs in "{0}" mode and this test will run '
                'only on "{1}" mode.'.format(settings_project, project)
            )
        else:
>           return func(*args, **kwargs)

robottelo/decorators/__init__.py:255: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
tests/foreman/ui/test_variables.py:451: in test_positive_update_type
    self.assertEqual(value, data['value'])
E   AssertionError: u'VWWDoJJMbc' != '954.960012424'
```

There might be an error during the update, but we probab ly don't assert for it. It might be type-specific, since the test has passed last build, where an array type has been used

--- a/.travis.yml
+++ b/.travis.yml
@@ -11,8 +11,8 @@ install:
 script:
     - flake8 .
     - make test-docstrings
-    - make docs
     - cp robottelo.properties.sample robottelo.properties
+    - make docs
     - tox
     # The `test-foreman-*` recipes require the presence of a Foreman
     # deployment, and they are lengthy. Don't run them on Travis.

--- a/robottelo/decorators/__init__.py
+++ b/robottelo/decorators/__init__.py
@@ -12,7 +12,8 @@ from robottelo.host_info import get_host_sat_version
 from robozilla.decorators import (  # noqa
     bz_bug_is_open, rm_bug_is_open,  # noqa
     skip_if_bug_open, _get_bugzilla_bug, _get_redmine_bug_status_id,  # noqa
-    _redmine_closed_issue_statuses  # noqa
+    _redmine_closed_issue_statuses,  # noqa
+    BugTypeError
 )
 
 LOGGER = logging.getLogger(__name__)
@@ -280,7 +281,46 @@ def config_picker():
     return config
 
 
+class run_in_one_thread_if_bug_open(skip_if_bug_open, object):
+    """A decorator that sets pytest marker and allows to select test that
+    should be run sequentially only if bug is open.
+    """
+
+    _wrapper = run_in_one_thread
+
+    def __call__(self, func):
+        """Return unchanged function or function decorated with
+        `pytest.mark.run_in_one_thread` marker decorator if bug is open.
+
+        :param func: The function being decorated.
+
+        :return: The return value of test method ``func``.
+        :raises BugTypeError: If ``bug_type`` is not recognized.
+        """
+        self.register_bug_id(func)
+
+        if self.bug_type not in ('bugzilla', 'redmine'):
+            raise BugTypeError(
+                '"{0}" is not a recognized bug type. Did you mean '
+                '"bugzilla" or "redmine"?'.format(self.bug_type)
+            )
+
+        if (self.bug_type == 'bugzilla' and bz_bug_is_open(
+                self.bug_id,
+                sat_version_picker=self.sat_version_picker,
+                config_picker=self.config_picker
+        )) or (self.bug_type == 'redmine' and rm_bug_is_open(self.bug_id)):
+            func = self._wrapper(func)
+        return func
+
+
 # Set the optional version and config pickers for robozilla decorators
+run_in_one_thread_if_bug_open = partial(
+    run_in_one_thread_if_bug_open,
+    sat_version_picker=get_host_sat_version,
+    config_picker=config_picker
+)
+
 bz_bug_is_open = partial(
     bz_bug_is_open,
     sat_version_picker=get_host_sat_version,

--- a/tests/foreman/ui/test_variables.py
+++ b/tests/foreman/ui/test_variables.py
@@ -30,7 +30,12 @@ from robottelo.datafactory import (
     invalid_names_list,
     valid_data_list,
 )
-from robottelo.decorators import run_only_on, tier1, tier2
+from robottelo.decorators import (
+    run_in_one_thread_if_bug_open,
+    run_only_on,
+    tier1,
+    tier2,
+)
 from robottelo.test import UITestCase
 from robottelo.ui.factory import make_smart_variable
 from robottelo.ui.locators import common_locators, locators, tab_locators
@@ -314,6 +319,7 @@ class SmartVariablesTestCase(UITestCase):
             self.assertIsNone(sv_value)
 
     @run_only_on('sat')
+    @run_in_one_thread_if_bug_open('bugzilla', 1440878)
     @tier1
     def test_positive_update_name(self):
         """Update Smart Variable name.
@@ -344,6 +350,7 @@ class SmartVariablesTestCase(UITestCase):
                     old_name = new_name  # for next iteration
 
     @run_only_on('sat')
+    @run_in_one_thread_if_bug_open('bugzilla', 1440878)
     @tier1
     def test_positive_update_variable_puppet_class(self):
         """Update Smart Variable puppet class.
@@ -411,6 +418,7 @@ class SmartVariablesTestCase(UITestCase):
             )
 
     @run_only_on('sat')
+    @run_in_one_thread_if_bug_open('bugzilla', 1440878)
     @tier1
     def test_positive_update_type(self):
         """Update Smart Variable with valid default value for all variable
@@ -456,6 +464,7 @@ class SmartVariablesTestCase(UITestCase):
                     self.assertEqual(value, data['value'])
 
     @run_only_on('sat')
+    @run_in_one_thread_if_bug_open('bugzilla', 1440878)
     @tier1
     def test_negative_update_type(self):
         """Attempt to update Smart Variable with invalid default value for all
@@ -1883,6 +1892,7 @@ class SmartVariablesTestCase(UITestCase):
             self.assertIn('masked-input', default_value.get_attribute('class'))
 
     @run_only_on('sat')
+    @run_in_one_thread_if_bug_open('bugzilla', 1440878)
     @tier1
     def test_positive_update_hidden_value(self):
         """Update the hidden default value of variable.

--- a/tests/robottelo/test_decorators.py
+++ b/tests/robottelo/test_decorators.py
@@ -1059,6 +1059,59 @@ class SkipIfBugOpen(TestCase):
         self.assertEqual(foo(), 42)
 
 
+class RunInOneThreadIfBugOpen(TestCase):
+    """Tests for :func:`robottelo.decorators.run_in_one_thread_if_bug_open`."""
+    def test_raise_bug_type_error(self):
+        with self.assertRaises(decorators.BugTypeError):
+            @decorators.run_in_one_thread_if_bug_open('notvalid', 123456)
+            def foo():
+                pass
+
+    @mock.patch('robottelo.decorators.bz_bug_is_open')
+    def test_mark_one_thread_bugzilla_bug(self, bz_bug_is_open):
+        bz_bug_is_open.side_effect = [True]
+
+        @decorators.run_in_one_thread_if_bug_open('bugzilla', 123456)
+        def foo():
+            return 42
+
+        self.assertTrue(hasattr(foo, 'run_in_one_thread'))
+        self.assertEqual(foo(), 42)
+
+    @mock.patch('robottelo.decorators.bz_bug_is_open')
+    def test_not_mark_one_thread_bugzilla_bug(self, bz_bug_is_open):
+        bz_bug_is_open.side_effect = [False]
+
+        @decorators.run_in_one_thread_if_bug_open('bugzilla', 123456)
+        def foo():
+            return 42
+
+        self.assertFalse(hasattr(foo, 'run_in_one_thread'))
+        self.assertEqual(foo(), 42)
+
+    @mock.patch('robottelo.decorators.rm_bug_is_open')
+    def test_mark_one_thread_redmine_bug(self, rm_bug_is_open):
+        rm_bug_is_open.side_effect = [True]
+
+        @decorators.run_in_one_thread_if_bug_open('redmine', 123456)
+        def foo():
+            return 42
+
+        self.assertTrue(hasattr(foo, 'run_in_one_thread'))
+        self.assertEqual(foo(), 42)
+
+    @mock.patch('robottelo.decorators.rm_bug_is_open')
+    def test_not_mark_one_thread_redmine_bug(self, rm_bug_is_open):
+        rm_bug_is_open.side_effect = [False]
+
+        @decorators.run_in_one_thread_if_bug_open('redmine', 123456)
+        def foo():
+            return 42
+
+        self.assertFalse(hasattr(foo, 'run_in_one_thread'))
+        self.assertEqual(foo(), 42)
+
+
 class SkipIfNotSetTestCase(TestCase):
     """Tests for :func:`robottelo.decorators.skip_if_not_set`."""
 

